Lib/test/test_typing.py

```diff
@@ -1790,14 +1785,11 @@ def test_dir(self):
 
     def test_cannot_subclass(self):
         with self.assertRaisesRegex(TypeError,
-                r'Cannot subclass typing\.Union'):
+                r"type 'types\.UnionType' is not an acceptable base type"):
             class C(Union):
                 pass
-        with self.assertRaisesRegex(TypeError, CANNOT_SUBCLASS_TYPE):
-            class C(type(Union)):
-                pass
         with self.assertRaisesRegex(TypeError,
-                r'Cannot subclass typing\.Union\[int, str\]'):
+                r"Union\[arg, \.\.\.\]: each arg must be a type\."):
```

Comment:
This seems like a weird error to get in this case. Both `int` and `str` are types.

Reply:
It's a consequence of the fact that subclassing something ends up calling the type of the base. Kind of like this:

```
>>> class C(42): pass
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: int() takes at most 2 arguments (3 given)
```

Possibly we could make a better error by overriding `__mro_entries__`, but not sure if that's worth it.

Reply:
I'm ok either way, but I'm inclined to feel it is worth it; it's something that someone might reasonably try, and the current error is very confusing, and it doesn't seem that hard to offer a better one.

Reply:
Added a `__mro_entries__` that always throws